### PR TITLE
DNL - h,cpp + unit tests

### PR DESCRIPTION
**Pull request by nanocoh**

### CODE REVIEW

File: test/dnl/DNLTests.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    SNLUniverse* univ = SNLUniverse::create();
+    SNLDB* db = SNLDB::create(univ);
+    SNLLibrary* library = SNLLibrary::create(db, SNLName("MYLIB"));
+    SNLDesign* mod = SNLDesign::create(library, SNLName("mod"));
+    univ->setTopDesign(mod);
+    auto inTerm = SNLScalarTerm::create(mod, SNLTerm::Direction::Input, SNLName("in"));
+    auto outTerm = SNLScalarTerm::create(mod, SNLTerm::Direction::Output, SNLName("out"));
+    //Create a DNL on top of the SNL
+    DNL* dnl = DNL::create();
+    assert(dnl != nullptr);
```
Comment (xtofalex) [inline test/dnl/DNLTests.cpp:54]: Use ASSERT_NE(dnl, nullptr);

File: src/dnl/DNL.h [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+typedef size_t DNLID;
+#define DNLID_MAX ((DNLID)-1)
+
+class DNL;
+class DNLTerminal;
+
+class DNLInstance {
+ public:
+    DNLInstance();
```
Comment (xtofalex) [inline src/dnl/DNL.h:32]: = default;

File: src/dnl/DNL.h [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    SNLInstTerm* getSnlTerm() const;
+    SNLBitTerm* getSnlBitTerm() const;
+    const DNLInstance& getDNLInstance() const;
+    bool isNull() const { return _id == (DNLID)DNLID_MAX; }
+    void setIsoID(DNLID isoID);
+    DNLID getIsoID() const;
+    bool isTopPort() const { return _terminal == nullptr; }
+
+ private:
+    SNLInstTerm* _terminal = nullptr;
```
Comment (xtofalex) [inline src/dnl/DNL.h:80]: private members with trailing _ not prefix as in SNL

File: src/dnl/DNL.h [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    SNLBitTerm* _bitTerminal = nullptr;
```
Comment (xtofalex) [inline src/dnl/DNL.h:81]: use the {} initialization:
SNLBitTerm* bitTerminal_ { nullptr };
for private members

**Review by xtofalex — COMMENTED** [review]
Already a series of feedback

File: test/dnl/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+INCLUDE(GoogleTest)
+
+SET(dnl_tests
+  DNLTests.cpp
+)
+
+ADD_EXECUTABLE(dnlTests ${dnl_tests})
+
+TARGET_LINK_LIBRARIES(dnlTests naja_snl naja_dnl gtest gmock gtest_main tbb)
```
[the remark this answers is not in the document]
Comment (nanocoh) [reply test/dnl/CMakeLists.txt:13]: What do you want to change here?

[the remark this answers is not in the document]
Comment (xtofalex) [reply test/dnl/CMakeLists.txt:14]: Nothing this is resolved by the target_include_directories in DNL CMakeLists.txt

File: src/dnl/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+add_library(naja_dnl SHARED ${DNL_SOURCES})
+
+target_link_libraries(naja_dnl PUBLIC naja_core naja_snl tbb tbbmalloc)
+#target_compile_options(naja_snl PRIVATE ${NAJA_CXX_WARNINGS}) 
+#target_compile_definitions(naja_snl PRIVATE SNL_DESTROY_DEBUG)
+
+target_include_directories(naja_dnl SYSTEM BEFORE PUBLIC ${Boost_INCLUDE_DIR})
+target_include_directories(naja_dnl SYSTEM BEFORE PUBLIC ${TBB_INCLUDE_DIR})
+target_include_directories(naja_dnl PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+set_target_properties(naja_dnl PROPERTIES PUBLIC_HEADER "${SNL_HEADERS}")
```
Comment (xtofalex) [inline src/dnl/CMakeLists.txt:21]: SNL_HEADERS => DNL_HEADERS

File: src/dnl/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  DNL.cpp
+)
+set(DNL_HEADERS
+  DNL.h
+)
+
+add_library(naja_dnl SHARED ${DNL_SOURCES})
+
+target_link_libraries(naja_dnl PUBLIC naja_core naja_snl tbb tbbmalloc)
+#target_compile_options(naja_snl PRIVATE ${NAJA_CXX_WARNINGS})
```
Comment (xtofalex) [inline src/dnl/CMakeLists.txt:15]: This you should uncomment with naja_snl => naja_dnl

**Pull request merged**